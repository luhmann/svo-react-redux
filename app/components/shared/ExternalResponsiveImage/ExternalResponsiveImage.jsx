import { React } from 'lib/commonImports.js'
import { buildResImageUrl } from 'lib/UrlBuilder.js'

import {RESPONSIVE_IMAGE_SIZES} from '../../../constants/Common.js'

const buildSrcSet = (objectId, sizes) => {
  return sizes
    .map((size) => (
      `${buildResImageUrl(objectId, size)} ${size}w`
    ))
    .join(', ')
}

const ExternalResponsiveImage = ({ img, className, srcset = RESPONSIVE_IMAGE_SIZES, sizes = '100vw' }) => (
  <img
    className={className}
    src={buildResImageUrl(img.objectId)}
    srcSet={buildSrcSet(img.objectId, srcset)}
    sizes={sizes}
  />
)

ExternalResponsiveImage.propTypes = {
  img: React.PropTypes.object.isRequired,
  className: React.PropTypes.string,
  srcset: React.PropTypes.array,
  sizes: React.PropTypes.string
}

export default ExternalResponsiveImage
