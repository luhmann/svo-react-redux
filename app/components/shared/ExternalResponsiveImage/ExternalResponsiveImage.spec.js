import React from 'react'
import renderer from 'react-test-renderer'

import ExternalResponsiveImage from './ExternalResponsiveImage.jsx'

describe.only('Component: <ExternalResponsiveImage />', () => {
  it('should render a responsive image', () => {
    const img = {
      objectId: 'lh1nmgmqugmlopswifqt',
      credits: 'Jan Florian Dietrich'
    }

    const component = renderer.create(
      <ExternalResponsiveImage img={img} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
