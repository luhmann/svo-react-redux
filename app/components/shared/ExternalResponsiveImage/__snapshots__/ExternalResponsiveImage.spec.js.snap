exports[`Component: <ExternalResponsiveImage /> should render a responsive image 1`] = `
<img
  className=""
  sizes="100vw"
  src="https://res.cloudinary.com/svo/image/upload/c_fit/lh1nmgmqugmlopswifqt"
  srcSet="https://res.cloudinary.com/svo/image/upload/c_fit,w_414/lh1nmgmqugmlopswifqt 414w, https://res.cloudinary.com/svo/image/upload/c_fit,w_1024/lh1nmgmqugmlopswifqt 1024w, https://res.cloudinary.com/svo/image/upload/c_fit,w_1280/lh1nmgmqugmlopswifqt 1280w, https://res.cloudinary.com/svo/image/upload/c_fit,w_1366/lh1nmgmqugmlopswifqt 1366w, https://res.cloudinary.com/svo/image/upload/c_fit,w_1920/lh1nmgmqugmlopswifqt 1920w, https://res.cloudinary.com/svo/image/upload/c_fit,w_2500/lh1nmgmqugmlopswifqt 2500w" />
`;
